app/src/store/cart.ts

```diff
@@ -386,26 +388,31 @@ export default function useCartStore() {
       const token = item.contributionToken;
       // collect the matching values for each grant in each round
       _predictions[item.grantId] = (grantRounds.value || []).map((round) => {
-        // all calculations are denominated in the rounds donationToken
-        const roundToken = round.donationToken;
-        // get the predictions for this grant in this round
-        const clr_predictions = getPredictionsForGrantInRound(item.grantId, grantRoundsCLRData.value[round.address]);
-        // no conversion is required if tokens are in the same currency
-        const contributionIsRoundToken = token.address == roundToken.address;
-        // if contribution/donationToken token is DAI we can skip that step of the conversion
-        const contributionIsDai = token.symbol === 'DAI';
-        const roundTokenIsDai = roundToken.symbol === 'DAI';
-        // take the initial contributionAmount and convert it to be denominated in donationToken
-        let amount = item.contributionAmount;
-        // convert amount to the donationToken (double hop to get into dai then into the donationToken)
-        amount = contributionIsDai || contributionIsRoundToken ? amount : getConvertedAmount(amount, token.address, 1);
-        amount =
-          roundTokenIsDai || contributionIsRoundToken ? amount : getConvertedAmount(amount, roundToken.address, -1);
-
-        const matching = getPredictedMatchingForAmount(
-          clr_predictions,
-          amount // pass in the donationToken denominated amount
-        );
+        let matching: number | boolean = false;
+        const metadata = grantRoundMetadata.value[round.metaPtr];
+        if (metadata && metadata.grants?.includes(parseInt(item.grantId))) {
+          // all calculations are denominated in the rounds donationToken
+          const roundToken = round.donationToken;
+          // get the predictions for this grant in this round
+          const clr_predictions = getPredictionsForGrantInRound(item.grantId, grantRoundsCLRData.value[round.address]);
+          // no conversion is required if tokens are in the same currency
+          const contributionIsRoundToken = token.address == roundToken.address;
+          // if contribution/donationToken token is DAI we can skip that step of the conversion
+          const contributionIsDai = token.symbol === 'DAI';
+          const roundTokenIsDai = roundToken.symbol === 'DAI';
+          // take the initial contributionAmount and convert it to be denominated in donationToken
+          let amount = item.contributionAmount;
+          // convert amount to the donationToken (double hop to get into dai then into the donationToken)
+          amount =
+            contributionIsDai || contributionIsRoundToken ? amount : getConvertedAmount(amount, token.address, 1);
+          amount =
+            roundTokenIsDai || contributionIsRoundToken ? amount : getConvertedAmount(amount, roundToken.address, -1);
```

Comment:
I don't follow the logic in this section. Why do we convert the amount in the `1` direction then back in the `-1` direction? I think the comment is saying regardless of token, we normalize the amount by first converting the price to DAI, then converting the DAI amount to be denominated in `donationToken`. But why go through DAI in the first place?

Reply:
This is for covering the situation where we need to convert the actual donation token in to the GrantRounds `donationToken` so that we can estimate with an approximation of the output of the swap (say we're contributing `UNI` but the rounds `donationToken` is set to `GTC`). 

I couldn't see another way to get into the desired `donationToken` other than adding every combination of path, is there an easier way that I'm missing?